### ISSUE TITLE
Error encountered while auto_delete with DelayedJob

### ISSUE THREAD

**Issue by chaadow** [algolia/algoliasearch-rails#75]
Hi,
I tried this snippet of code as suggested in the readme for enqueuing Algolia operations  : 
```ruby
class Contact < ActiveRecord::Base
  include AlgoliaSearch

  algoliasearch enqueue: :trigger_delayed_job do
    attribute :first_name, :last_name, :email
  end

  def self.trigger_delayed_job(record, remove)
    if remove
      record.delay.remove_from_index!
    else
      record.delay.index!
    end
  end
end
```
My application is running rails 3.2.22 with ruby 2.2.2 and DelayedJob 4.0.6

Without delayed job, the auto_remove works perfectly fine.
However, when I try to remove the project with the enqueuing system set, I get the following error : 
```
ArgumentError:
       job cannot be created for non-persisted record:
```
after further research, I found this DJ related issue : https://github.com/collectiveidea/delayed_job/issues/820
Is there any workaround to this, since I want to remove my model asynchronously from Algolia only after my project has been deleted

**Comment by redox** [comment]
That's weird, do you have the associated backtrace? Is it raised from the `record.delay.{remove_from_,}index!` methods?

**Comment by chaadow** [comment]
Yes it is raised from `record.delay.remove_from_index!` When I destroy my model. For example : `project.destroy`

**Comment by chaadow** [comment]
@redox Here is an example of a failing spec backtrace : 
```
 Failure/Error: expect { delete :destroy, :id => @project }.to change(Project, :count).by(-1)
     ArgumentError:
       job cannot be created for non-persisted record: #<Project

```

**Comment by redox** [comment]
Can you just try to run a `record.remove_from_index!` and another `record.delay.remove_from_index!` from the rails console, to check if you're experiencing the same? I'm don't get it right now :/

**Comment by chaadow** [comment]
`record.remove_from_index!` works whereas `record.delay.remove_from_index!` doesn't.

**Comment by chaadow** [comment]
Actually, it doesn't work only after I destroy the record and the `self.trigger_delayed_job` is triggered.
Since it's a "remove" it steps in the `if remove` condition, and finally the exception I told you about above is raised.
When I try without the "delay" it works whereas with the delay it doesn't and it displays this error :  
```
ArgumentError:
       job cannot be created for non-persisted record: #<Project
```

**Comment by redox** [comment]
Could you just try with another (non-algolia) method? From what I understand, your `Project` object cannot be serialized but I don't get why it could be related to the `remove_from_index!`.

**Comment by chaadow** [comment]
I am only using what's suggested in the readme for delayedjob.  `record.delay.remove_from_index!` works outside the `self.trigger_delayed_job`, if I try it in the console. However when I try it after a destroy,  for instance :  `project.destroy`, then as I explained,`record.delay.remove_from_index!` does not work anymore.
Could you try that to see maybe you have to update the example you're suggesting in the readme ?

**Comment by chrisyeung1121** [comment]
+1. I am having the same problem.

**Comment by drewtunney** [comment]
I ran into the same issue. The workaround I've used is to delete the record by ID from the index in the background job. It seems to work fine

```ruby
 def trigger_delayed_job(record, remove)
      if remove
        index = Algolia::Index.new("#{self.name}_#{Rails.env}")
        index.delete_object(record.uuid)
      else
        obj = self.find(record.id)
        obj.index!
      end
    end

```
